pandas/core/algorithms.py

```diff
@@ -556,7 +556,7 @@ def factorize(
 
     >>> codes, uniques = pd.factorize(['b', 'b', 'a', 'c', 'b'])
     >>> codes
-    array([0, 0, 1, 2, 0])
+    array([0, 0, 1, 2, 0], dtype=int64)
```

Comment:
Whether this is displayed by numpy depends on your machine / OS, I think (eg on my linux laptop, it is not displayed). So we cannot simply add it. If we want to have the doctests pass locally in such cases, it probably needs some `...`

Reply:
Hello @jorisvandenbossche! 
Could you elaborate more on `...`? 
So actually it is not really problem because the difference arises due to OS running?

Reply:
Well, it's still a problem for people who want to run it locally and then see errors (like you did I suppose).

About the ..., see https://docs.python.org/3/library/doctest.html#doctest.ELLIPSIS

I am not fully sure how we deal with this in other places (cc @datapythonista  ?)

Reply:
It's surprising that numpy changes the `__repr__` depending on the platform. I would report it to them, may be there is a good reason, or may be it's just some old code.

For fixing our doctrsts, I'd check if the same happens with `__str__`, and use `print` if it doesn't. Otherwise, I'd simply add an ignore. It's not a great solution, since it's shown to the user, but it's what we've been doing. Better than leave failing tests IMO.

Reply:
I think it is a well known issue with numpy and doctests (I think, not fully sure, but eg in scikit-learn we regularly ran into similar issues as well). I think it specifically comes up here because it are ints (and if you are on a platform where int64 is not the default, it shows the dtype).

Using

```
array([0, 0, 1, 2, 0]...)
```

should work I think. @farhanreynaldo can you try if what this it passes locally for you?

Using `print(..)` is indeed another option (print never shows the dtype).

Reply:
Ah, I see, that makes more sense. Missed the `...` sorry, if that works surely better than the skip.

Reply:
@jorisvandenbossche @datapythonista Sure, I'll try that out and commit if it's working out

Reply:
I have made the changes using ellipsis but the test still throws an error. After doing a quick look, it seems like we need to enable `optionflags=doctest.ELLIPSIS`.

Reply:
You can add that to `doctest_optionflags` in setup.cfg, I think

Reply:
The problem still arises eventhough I have added ellipsis to `doctest_optionflags`. Am I missing something here?

![image](https://user-images.githubusercontent.com/8732795/77422970-f81e1d80-6e00-11ea-90a3-ef5894ab9621.png)

Reply:
This looks correct, if you update this PR with the changes to `setup.cfg` and adding the `...`, we can have a look and give it a try. But seems like it should be working.

Reply:
For me this works locally, but I am also on a 64bit machine, so that might not be a good indication.